Mathlib/AlgebraicGeometry/Morphisms/UnderlyingMap.lean

```diff
@@ -137,10 +131,47 @@ instance : (topologically ClosedEmbedding).RespectsIso :=
   topologically_respectsIso _ (fun e ↦ e.closedEmbedding) (fun _ _ hf hg ↦ hg.comp hf)
 
 instance closedEmbedding_isLocalAtTarget : IsLocalAtTarget (topologically ClosedEmbedding) :=
-  topologically_isLocalAtTarget _
-    (fun _ s hf ↦ hf.restrictPreimage s)
-    (fun _ _ _ hU hfcont hf ↦ (closedEmbedding_iff_closedEmbedding_of_iSup_eq_top hU hfcont).mpr hf)
+  topologically_isLocalAtTarget' _ fun _ _ _ ↦ closedEmbedding_iff_closedEmbedding_of_iSup_eq_top
 
 end ClosedEmbedding
 
+section Dominant
+
+variable {X Y Z : Scheme.{u}} (f : X ⟶ Y) (g : Y ⟶ Z)
+
+/-- A morphism of schemes is dominant if the underlying map is. -/
+@[mk_iff]
+class Dominant : Prop where
+  denseRange : DenseRange f.base
+
+lemma dominant_eq_topologically :
+    @Dominant = topologically DenseRange := by ext; exact dominant_iff _
+
+lemma Scheme.Hom.denseRange (f : X.Hom Y) [Dominant f] : DenseRange f.base :=
+  Dominant.denseRange
+
+instance (priority := 100) [Surjective f] : Dominant f := ⟨f.surjective.denseRange⟩
+
+instance [Dominant f] [Dominant g] : Dominant (f ≫ g) := ⟨g.denseRange.comp f.denseRange g.base.2⟩
+
+instance : MorphismProperty.IsStableUnderComposition @Dominant := ⟨fun _ _ _ _ ↦ inferInstance⟩
+
+lemma Dominant.of_comp [H : Dominant (f ≫ g)] : Dominant g := by
+  rw [dominant_iff, denseRange_iff_closure_range, ← Set.univ_subset_iff] at H ⊢
+  exact H.trans (closure_mono (Set.range_comp_subset_range f.base g.base))
+
+lemma Dominant.comp_iff [Dominant f] : Dominant (f ≫ g) ↔ Dominant g :=
+  ⟨fun _ ↦ of_comp f g, fun _ ↦ inferInstance⟩
+
+instance : MorphismProperty.RespectsIso @Dominant :=
+  MorphismProperty.respectsIso_of_isStableUnderComposition fun _ _ f (_ : IsIso f) ↦ inferInstance
+
+instance dominant_isLocalAtTarget : IsLocalAtTarget @Dominant :=
+  have : MorphismProperty.RespectsIso (topologically DenseRange) :=
+    dominant_eq_topologically ▸ instRespectsIsoSchemeDominant
```

Comment:
I think it is safer to give the `RespectsIso` instance above a name and use it here.